tests/infra/jsonrpc.py

```diff
@@ -54,19 +54,23 @@ def truncate(string, max_len=256):
 
 
 class Request:
-    def __init__(self, id, method, params, jsonrpc="2.0"):
+    def __init__(self, id, method, params, readonly_hint=None, jsonrpc="2.0"):
```

Comment:
I think I would prefer hint=Hint.W, with an enum Hint of W, MW, R. We could even test the hints that way (and we probably should!).

Reply:
The RO hint, if it exists, is only `True` or `False` so I don't think that we need an enum for this. OTOH, the handler on the server side is installed as `Write`, `Read` or `MayWrite`.

Reply:
Ok, so should it default to False then? Why None?